src/components/CardList.js

```diff
@@ -0,0 +1,23 @@
+import '../styles/CardList.css';
+import Card from '../components/Card';
+
+const CardList = ({ folderInfo }) => {
+  const cardList = folderInfo?.folder.links;
+  console.log(cardList);
+
+  return (
+    <div className="cardlist">
+      {cardList ? (
+        <>
+          {cardList.map((cardlist) => (
+            <Card cardlist={cardlist} key={cardlist.id} />
+          ))}
+        </>
+      ) : (
+        <div>폴더가 비어있습니다.</div>
+      )}
+    </div>
+  );
+}
```

Comment:
`CardList`로 오는 데이터인 `folderInfo`를 보면 `folder`페이지에서 데이터를 가져와서 `folder`-> `folderContent` -> `CardList` 순서대로 데이터가 흐르고 있죠?  하지만 `folderInfo` 데이터는 `folder`, `folderContent`에서는 아무런 역할을 하지 않고 그냥 데이터를 흘려주기만 할 뿐입니다. 이런경우에는 차라리 데이터가 필요한 컴포넌트에서 직접 데이터를 가져오는편이 좋습니다.  데이터를 props로 여러 단계의 컴포넌트를 거쳐서 흘려주기만 할때  props drilling라고 합니다.

```suggestion
const CardList = () => {
  const [folder, setFolder] = useState({
    folder: {
      links: [],
    },
  });

  useEffect(() => {
    const handleLoad = async () => {
      try {
        const folder = await getFolder();
        setFolder(folder);
      } catch (error) {
        console.error(error);
      }
    };
    handleLoad();
  }, []);

  const cardList = folder?.folder.links;

  return (
    <div className="cardlist">
      {cardList ? (
        <>
          {cardList.map(card => (
            <Card cardlist={card} key={card.id} />
          ))}
        </>
      ) : (
        <div>폴더가 비어있습니다.</div>
      )}
    </div>
  );
};
```

Reply:
d2b06928a94e4648aaba03d17571cdd1b00950b9 `children` 을 사용해서 props drilling을 해결해봤습니도 !